### PR TITLE
Add a pipeline spec and some miscellaneous docs

### LINKED ISSUE #332: Flash-crowd behavior
It's been observed in my cluster that the following scenario occurs:

I have a non-reduce pipeline with the degree of parallelism set to N.  An input commit comes in.  The input commit contains a number of large files, each of which contains more than N blocks.

Now, the way we assign inputs to jobs, the n-th job will get the n-th block to every file.  So the first job gets the first block to every file, the second job gets the second block to every file, etc.

Since each file contains more than N blocks, each job essentially sees the same files, i.e. all files.

What happens next is that, each job starts processing the same file, i.e. the first file in the input directory.  Since we shard traffic by file path, all these jobs start reading file from the same server.  So even though we'd ideally want traffic to spread across the cluster, we in fact get a flash-crowd behavior where all jobs talk to the same server.

### PR DESCRIPTION
**Pull request by derekchiang**
Fixes #332 and checks "pipeline spec guide" and "Docs for delete/overwrite behavior" in #309

### CODE REVIEW

**Comment by derekchiang** [comment]
@jdoliner review plz?

**Comment by jdoliner** [comment]
@JoeyZwicker you mind reviewing this one?

File: doc/pipeline_spec.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        "grep orange /pfs/data/* >/pfs/out/orange"
+    ]
+  },
+  "parallelism": "4",
+  "inputs": [
+    {
+      "repo": {
+        "name": "data"
+      },
+      "reduce": true
```
Comment (JoeyZwicker) [inline doc/pipeline_spec.md]: Does it make sense for filter to be a reduce? I feel like this is clearly a map job. I think the average use case is someone using their own container, not bash. Can we make the image and cmd something more demonstrative? The quickstart guide already has the bash shell example covered.

**Comment by JoeyZwicker** [comment]
Except for potentially changing the example, everything looks good.

**Comment by derekchiang** [comment]
@JoeyZwicker changed the example.  Another look?

**Comment by JoeyZwicker** [comment]
We discussed potential other tweaks the example offline. Let's merge this for now and we can always change the example a bit with feedback from @jdoliner

**Comment by jdoliner** [comment]
This LGTM.

**Pull request merged**